### PR TITLE
cni: annotate NAD with host interface name

### PR DESCRIPTION
**Pull request by privateip**
Annotate the NetworkAttachmentDefinition with the deterministic host-side interface name under \`k8s.v1.cni.cncf.io/host-interface\`. This allows downstream processes to discover the interface name before pod launch without querying VPCAttachment CRs.

**Changes:**
- New \`internal/cni/nad.go\` with \`annotateNAD()\` and \`parsePodNamespace()\`
- Unit tests for \`parsePodNamespace\` covering edge cases
- RBAC rule for patching \`network-attachment-definitions\`
- Best-effort integration in \`cmdAdd\` — tolerates NotFound and Conflict
- Deduplicate k8s client init by storing on \`resourceTracker\`

**Why:** Interface name is deterministic from VPC+VPCAttachment but downstream processes need a queryable resource. Annotating the NAD is cleaner than creating a dedicated CRD for this single identifier.

Generated with Qwen Code (1-shotted by Qwen-Coder)

### CODE REVIEW

File: internal/cni/nad.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	err := k8s.Patch(ctx, nad, client.RawPatch(types.JSONPatchType, []byte(patch)))
+	if err != nil {
+		// Tolerate not-found (NAD managed by external operator that may not
+		// have created it yet) and conflict (annotation already exists from
+		// a previous invocation).
+		if !apierrors.IsNotFound(err) && !apierrors.IsConflict(err) {
+			slog.Warn("annotate NAD: patch failed",
+				"name", nadName, "namespace", nadNamespace, "err", err)
+		}
+		return
```
Comment (scotwells) [inline internal/cni/nad.go]: Why wouldn't this hard fail?

Comment (privateip) [reply internal/cni/nad.go]: Right now its possible that the NAD isn't there and it probably should fail hard.   I'm working on a new PR that will automate NAD creation so technically this should never happen.    That said, its probably good practice to force this scenario to hard fail.

Comment (privateip) [reply internal/cni/nad.go]: fixed in 8ef8fa3

**Review by scotwells — APPROVED** [review]
